src/python/pants/bin/daemon_pants_runner.py

```diff
@@ -25,7 +24,7 @@ class ExclusiveRequestTimeout(Exception):
     """Represents a timeout while waiting for another request to complete."""
 
 
-class DaemonPantsRunner(RawFdRunner):
+class DaemonPantsRunner:
```

Comment:
Two reasons I don't think we should not do that:

1. Now that the protocol is defined in `native_engine.pyi`, we can't import it. I couldn't think of a good spot to define that protocol in a `.py` file.
2. It was an abuse of `Protocol`. You should not subclass `Protocol`, and in fact, doing so was hiding a deviation from our expected API that the protocol said `stdin_fd` instead of `stdin_fileno`.